src/main/java/org/apache/sling/api/resource/ResourceUtil.java

```diff
@@ -271,12 +271,19 @@ public static String getParent(final String path, final int level) {
      * @throws NullPointerException If <code>path</code> is <code>null</code>.
      */
     public static @NotNull String getName(@NotNull String path) {
+        if (path == null) {
+            throw new NullPointerException("provided path is null");
+        }
         if ("/".equals(path)) {
             return "";
         }
 
         // normalize path (remove . and ..)
         path = normalize(path);
+        if (path == null) {
+            throw new IllegalArgumentException(
+                    String.format("normalized path '%s' resolves to a path higher than root", path));
```

Comment:
path is already null when the exception message is created

Reply:
good spot.